sentry-core/src/clientoptions.rs

```diff
@@ -91,6 +116,8 @@ pub struct ClientOptions {
     /// is started at the time of `sentry::init`, and will persist for the
     /// application lifetime.
     pub auto_session_tracking: bool,
+    /// Determine how Sessions are being tracked.
+    pub session_mode: SessionMode,
```

Comment:
Could we consider combining this with the session tracking flag into a `session_mode: Application | Request | Off`?

Reply:
I thought about this a bit, and a tri-state would not work when you don’t want to have auto-tracking, but want to manually track request-mode sessions.

Also related to your other question: Request-Mode applies to all sessions, I updated the doc comment to accordingly. So choosing between application/request-mode basically means that you make a choice if you want aggregation at the cost of losing durations, or not.